fiftyone/core/collections.py

```diff
@@ -961,9 +982,13 @@ def tail(self, num_samples=3):
             num_samples (3): the number of samples
 
         Returns:
-            a list of :class:`fiftyone.core.sample.Sample` objects
+            a list of :class:`fiftyone.core.sample.Sample` or
+            :class:`fiftyone.core.sample.SampleView` objects
         """
-        return [s for s in self[-num_samples:]]
+        cursor = self._get_first(reverse=True, limit=num_samples)
+        samples = [self._make_sample(d) for d in cursor]
+        samples.reverse()
```

Comment:
`reverse()` here so that sample order is maintained, for backwards compatibility

Reply:
Good catch 👍